src/k8s/pkg/k8sd/features/contour/ingress.go

```diff
@@ -0,0 +1,118 @@
+// ApplyIngress will install a delegation resource via helm chart
+// for the default TLS secret if ingress.DefaultTLSSecret is set.
+// ApplyIngress returns an error if anything fails.
+// Contour CRDS are applied through a ck-contour common chart (Overlap with gateway)
+func ApplyIngress(ctx context.Context, snap snap.Snap, ingress types.Ingress, _ types.Network, _ types.Annotations) error {
+	m := snap.HelmClient()
+
+	if !ingress.GetEnabled() {
+		if _, err := m.Apply(ctx, chartContour, helm.StateDeleted, nil); err != nil {
+			return fmt.Errorf("failed to uninstall ingress: %w", err)
+		}
+	}
+
+	// Apply common contour CRDS, these are shared with gateway
+	if err := applyCommonContourCRDS(ctx, snap, true); err != nil {
+		return fmt.Errorf("failed to apply common contour CRDS: %w", err)
+	}
+
+	var values map[string]any
+	values = map[string]any{
+		"envoy-service-namespace": "projectcontour",
+		"envoy-service-name":      "envoy",
+		"contour": map[string]any{
+			"manageCRDs": false,
+			"ingressClass": map[string]any{
+				"name":    "ck-ingress",
+				"create":  true,
+				"default": true,
+			},
+		},
+	}
+
+	if ingress.GetEnableProxyProtocol() {
+		contour := values["contour"].(map[string]any)
+		contour["extraArgs"] = []string{"--use-proxy-protocol"}
+
+	}
+
+	changed, err := m.Apply(ctx, chartContour, helm.StatePresent, values)
+	if err != nil {
+		return fmt.Errorf("failed to enable ingress: %w", err)
+	}
+
+	if changed {
+		if err := rolloutRestartContour(ctx, snap, 3); err != nil {
+			return fmt.Errorf("failed to rollout restart contour to apply ingress: %w", err)
+		}
+	}
+
+	// Install the delegation resource for the default TLS secret.
+	// The default TLS secret is created by the user created,
+	// and gets set via k8s set defaultTLSSecret=bananas.
+	if ingress.GetDefaultTLSSecret() != "" {
+		values = map[string]any{
+			"defaultTLSSecret": ingress.GetDefaultTLSSecret(),
+		}
+		if _, err := m.Apply(ctx, chartDefaultTLS, helm.StatePresent, values); err != nil {
+			return fmt.Errorf("failed to install the delegation resource for default TLS secret: %w", err)
+		}
+	} else {
+		if _, err := m.Apply(ctx, chartDefaultTLS, helm.StateDeleted, nil); err != nil {
+			return fmt.Errorf("failed to uninstall the delegation resource for default TLS secret: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// applyCommonContourCRDS will install the common contour CRDS when enabled is true.
+// These CRDS are shared between the contour ingress and the gateway feature.
+func applyCommonContourCRDS(ctx context.Context, snap snap.Snap, enabled bool) error {
+	m := snap.HelmClient()
+	if enabled {
+		if _, err := m.Apply(ctx, chartCommonContourCRDS, helm.StatePresent, nil); err != nil {
+			return fmt.Errorf("failed to install common CRDS: %w", err)
+		}
+	} else {
+		if _, err := m.Apply(ctx, chartCommonContourCRDS, helm.StateDeleted, nil); err != nil {
+			return fmt.Errorf("failed to uninstall common CRDS: %w", err)
+		}
+	}
```

Comment:
What do you think of this? Often, when an if's only other branch is else, the else can be removed.

```suggestion
	if enabled {
		if _, err := m.Apply(ctx, chartCommonContourCRDS, helm.StatePresent, nil); err != nil {
			return fmt.Errorf("failed to install common CRDS: %w", err)
		}
		return nil
	}
	
    if _, err := m.Apply(ctx, chartCommonContourCRDS, helm.StateDeleted, nil); err != nil {
	    return fmt.Errorf("failed to uninstall common CRDS: %w", err)
    }
    
	return nil
```